src/Symfony/Component/Messenger/Middleware/HandleMessageMiddleware.php

```diff
@@ -58,6 +63,10 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
                 continue;
             }
 
+            $this->eventDispatcher?->dispatch(new HandlerStartingEvent($envelope, $handlerDescriptor->getName()));
```

Comment:
It would be great if the `$handlerDescriptor` is fully passed to this event, and that a subscriber can mutate it.

This would solve my use case that I tried to implement in:
- https://github.com/symfony/symfony/pull/50980

It would allow event subscribers to:
- read everything from the HandlerDescriptor, so also the options 
- set a new HandlerDescriptor that has a custom callable that does specific logic.

Reply:
I'm all for adding the whole `HandlerDescriptor` in the events, you're right that it opens up a whole lot of possibilities.

However, I have no idea of the potential consequences of live-replacing a handler, so I'll leave it to others to decide if making the `HandlerDescriptor` replaceable is the way to go!

Reply:
I'm all for adding the whole `HandlerDescriptor` in the events, you're right that it opens up a whole lot of possibilities.

However, I have no idea of the potential consequences of live-replacing a handler, so I'll leave it to others to decide if making the `HandlerDescriptor` replaceable is the way to go!

Reply:
@ro0NL @OskarStark What do you think?

Reply:
I have to admit that I also don't know what consequences this can have

Reply:
@ruudk Do you actually have a need to replace the `HandlerDescriptor` at the moment, or can we leave it for a later PR to maybe make it replaceable in the future?

Reply:
Having it would be great, but maybe I can use the event system already to do what I want. I agree that we can later always decide to open this up.

Could we at least add the full HandlerDescriptor instead of only the name? That would make it a bit better.

Reply:
> Could we at least add the full HandlerDescriptor instead of only the name? That would make it a bit better.

Yes, definitely. I'll change this.

Reply:
@ruudk events now have the full `HandlerDescriptor`!